flink-yarn/src/main/java/org/apache/flink/yarn/YarnConfigKeys.java

```diff
@@ -36,7 +36,10 @@
 	public static final String FLINK_JAR_PATH = "_FLINK_JAR_PATH"; // the Flink jar resource location (in HDFS).
 	public static final String FLINK_YARN_FILES = "_FLINK_YARN_FILES"; // the root directory for all yarn application files
 
+	@Deprecated
 	public static final String KEYTAB_PATH = "_KEYTAB_PATH";
+	public static final String REMOTE_KEYTAB_PATH = "_REMOTE_KEYTAB_PATH";
```

Comment:
Is it actually necessary to store these keys in the environment, can't we directly use the Flink `Configuration` to retrieve these settings?

Reply:
This is a good catch (to also response to other comment --> technically speaking there should only be one ConfigKey in the JM/TM perspective: the "REMOTE_KEYTAB_PATH", or previously known as "KEYTAB_PATH". 

I would revert the changes I've done here. since if the keytab is shipped over as you mentioned, it doesn't matter whether it is originated from a remote ship location or pre-installed.